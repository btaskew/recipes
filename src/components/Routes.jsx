import React from 'react';

import {BrowserRouter, Route, Switch} from 'react-router-dom';
import IngredientPage from 'ingredientPage/IngredientPage';
import MainMenu from 'mainMenu/MainMenu';
import RecipeSearchPage from 'recipeSearch/RecipeSearchPage';
import AddRecipeContainer from 'addRecipePage/AddRecipeContainer';
import RecipePageContainer from 'recipePage/RecipePageContainer';
import NoRoute from 'components/NoRoute';

function Routes() {
    return (
        <BrowserRouter>
            <React.Fragment>
                <Switch>
                    <Route path="/" exact component={MainMenu} />
                    <Route path="/add" exact component={AddRecipeContainer} />
                    <Route path="/recipe/:id" exact component={RecipePageContainer} />
                    <Route path="/search" exact component={RecipeSearchPage} />
                    <Route path="/ingredient/:name" exact component={IngredientPage} />
                    <Route component={NoRoute} />
                </Switch>
            </React.Fragment>
        </BrowserRouter>
    );
}

export default Routes;
