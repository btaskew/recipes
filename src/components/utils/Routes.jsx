import React from 'react';

import {BrowserRouter, Route} from 'react-router-dom';
import MainMenu from 'pages/MainMenu';
import SearchPage from 'pages/SearchPage';
import AddRecipe from 'pages/AddRecipe';
import RecipePage from 'pages/RecipePage';
import IngredientPage from 'pages/IngredientPage';

function Routes() {
    return (
        <BrowserRouter>
            <div>
                <Route path="/" exact component={MainMenu} />
                <Route path="/search" component={SearchPage} />
                <Route path="/add" component={AddRecipe} />
                <Route path="/recipe/:id/:name" component={RecipePage} />
                <Route path="/ingredient/:id/:name" component={IngredientPage} />
            </div>
        </BrowserRouter>
    );
}

export default Routes;
