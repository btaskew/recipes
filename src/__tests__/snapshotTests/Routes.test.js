import React from 'react';
import renderer from 'react-test-renderer';

import Routes from 'components/Routes';
import {MemoryRouter} from 'react-router';

describe('Routes', () => {
    it('renders correctly', () => {
        const routes = renderer
            .create(
                <MemoryRouter>
                    <Routes  />
                </MemoryRouter>
            )
            .toJSON();
        expect(routes).toMatchSnapshot();
    });
});
