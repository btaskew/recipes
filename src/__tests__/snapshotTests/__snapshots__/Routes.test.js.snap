// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes renders correctly 1`] = `
Array [
  <div
    className="heading modal-header"
  >
    <h4
      className="modal-title"
    >
      No route found!
      <span
        className="float-right glyphicon glyphicon-home"
        href="/"
        onClick={[Function]}
        style={undefined}
      />
    </h4>
  </div>,
  <div
    className="modal-body"
  >
    <div
      className="alert alert-danger"
      role="alert"
    >
      404 - No route found!
    </div>
  </div>,
]
`;
